.cirrus.yml

```diff
@@ -161,11 +172,13 @@ task:
 
 task:
   name: 'ASan + LSan + UBSan + integer, no depends, USDT'
+  # Skip if no Noble worker exists (for a fork)
```

Comment:
It stays pending, at least in the Cirrus interface.

Reply:
Trying that UX in https://github.com/Sjors/bitcoin/pull/36 ...

Reply:
On the first push of a branch, due to the UI glitched described below (https://github.com/bitcoin/bitcoin/pull/29274#issuecomment-1967030916) they show as skipped:

<img width="875" alt="Scherm­afbeelding 2024-02-27 om 17 40 10" src="https://github.com/bitcoin/bitcoin/assets/10217/6a76f6f1-25dc-4235-b04b-0e605da8f7fc">

That seems fine. On the second push I expect them to stay "in progress" forever. Will add a screenshot later.

I agree that hiding the task altogether is not ideal (as ecf01f42c4df4193098e1edb7a8e19c42677d2ef does). Perhaps marking it as skipped is better.

Reply:
Just generally, I'd rather not add more code/complication to our CI infra to work around quirks/annoying side-effects in other infra, especially when the number of people affected by them are very close to 0.

Reply:
>  it's currently quite tedious to run CI on forks,, which encourages people to make pull requests to the main repo that could be avoided.

Maybe, but anyone can currently run any of the CI jobs they'd like, locally, against any code they'd like (not sure what forks, or nested PRs has to do with it, as long as there is a git branch with our ci infra, the CI job can run?). It seems that the requirement here is to have a Cirrus/GitHub interface?

Reply:
It's not that many lines really (apart from documentation), especially as @maflcko has been chipping away at the commits.

A few more lines can be dropped when Ubuntu 24.04 is out. efc22a9ae7b4d1b346e5f1ac8578927a27286917 can be dropped in a followup if the underlying issue is solved.

I don't mind giving people the option to use Microsoft, but that doesn't remove the need to support self-hosting imo.

> the requirement here is to have a Cirrus/GitHub interface?

Yes, for the same reason we have that on the main repo: it's a much lower barrier to entry. It helps the author by catching mistakes early and it gives reviewers the option to avoid wasting time on things that don't pass CI.

The use case I have in mind is someone who wants to improve the Template Provider in #29432 (before it's merged). I have dozens of TODO's in that PR which I'd like help with. In order to get the benefit of a full CI run, they currently have to open a PR to the main repo. With the tweaks in my PR they can simply open a PR to my fork instead. I think that's better for everyone.

Reply:
Well for one thing it's _way_ faster.

Reply:
It's also not this black and white. Sure, Microsoft and Cirrus can screw us over by faking the CI result. But by not hosting them for the machines themselves, it's both easier to catch them doing them than and easier to move elsewhere.

Reply:
If I run a self-hosted CI I can see what it's doing. If it fails on my machine and is marked as successful on Github, I can see that. If I collude with Github that's a different story of course.